### PR TITLE
Assurances and Compliance error messages

### LINKED ISSUE #3358: Add inline validation for Assurances and Compliance
Add inline validation for Assurances and Compliance

*These validations will be triggered by the error check that is currently being tested. #3488

|Field|Content|Considerations|
|---|---|---|
|all radio buttons, If blank|Select yes or no.||
|all radio buttons, if no > Please explain|Provide an explanation.||

**Mockup:**. (How error text would fall, underneath the field.)

### This task is done when...

- [ ] all radio buttons have inline validation

### PR DESCRIPTION
**Pull request by thetif**
Resolves #3358 

### Description
Adds error messages to assurances and compliance. If yes or no isn't selected, an error message of "Select yes or no" is displayed under the radio button. If no is selected but there is no explanation, then "Provide an explanation" is displayed under the text box.

### Automated test cases written
Updated the unit test to use react-testing-library instead of enzyme.

| Test Type | Given | When | Then |
|---|---|----|---|
| unit | adminCheck is off | page loads | no errors are displayed |
| unit | adminCheck is on | page loads | displays "Select yes or no" for all of unanswered radio buttons |
| unit | adminCheck is on | yes is selected for the radio button | the radio error goes away |
| unit | adminCheck is on | no is selected | the radio button error goes away, the explanation text area is displayed, and "Provide an explanation" is displayed under the text area |
| unit | adminCheck is on | no was selected and then yes is selected again | the radio error goes away |
| unit | adminCheck is on | no is selected and an explanation is added | the radio error and the text area error go away |
| e2e | an empty APD | the page loads | displays "Select yes or no" for all 15 radio buttons |
| e2e | an empty APD | the user selects values for all 15 radio buttons | all of the error messages go away

### Steps to manually verify this change

1. create an APD
2. navigate to Assurances and Compliance
3. confirm that all of the radio buttons display error message
4. select yes on one of the radio buttons
5. confirm that the error message goes away
6. select no on one of the radio buttons
7. confirm that the text area opens up and displays an error message
8. select yes on the same radio button
9. confirm that the error message goes away again
10. select no on the same radio button again
11. enter text in the text area that opens
12. confirm that the text area error message goes away

### This pull request is ready to review when

- [x] Automated tests are updated (and all tests are passing)
- [x] Pull request has been labeled, if applicable with feature, content, bug, tests, refactor
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when

- [x] Code has been reviewed by someone other than the original author
- [x] QA has verified the functionality related to the change
- [x] QA has verified the accessibility using tools such as screen readers and WAVE (accessibility evaluation tool)
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Review by Sun-Mountain — APPROVED** [review]
⭐

File: web/src/pages/apd/ApdReadOnly.test.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ const setup = (props = {}, options = {}) => {
   };
 };
 
+jest.setTimeout(300000);
```
Comment (tbolt) [inline web/src/pages/apd/ApdReadOnly.test.js:26]: still need this?

**Review by tbolt — APPROVED** [review]

File: web/src/pages/apd/ApdReadOnly.test.js (hunk 1 above)
Comment (thetif) [reply web/src/pages/apd/ApdReadOnly.test.js:26]: yes, unfortunately :(

**Comment by mirano-darren** [comment]
Nice! Looks good to me. Note: Admin check is on by default
| Given  | When | Then | Result |
| ------------- | ------------- | ------------- | ------------- |
| Assurances and Compliance radio buttons  |  Admin check is on  | Error message appears if no option is selected. Error message goes away once a selection is made.  | PASS ✅ |
| Please Explain text box  | Admin check is on, and field is empty/filled out  | Error message appears if empty, error message goes away if filled out.  | PASS ✅ |
| Assurances and Compliance page | Running WAVE on the page| No accessibility errors were found | PASS ✅ |
| Assurances and Compliance page | Using NVDA Screen reader | Page was read out correctly, all errors were read out loud | PASS ✅ |

**Review by akuas — APPROVED** [review]
All the error message behaviors work as expected.

**Pull request merged**